test/runtests.jl

```diff
@@ -203,6 +217,32 @@ const url = "http://127.0.0.1:$port"
         end
     end
 
+    @testset "task backtraces" begin
+        done = Threads.Atomic{Bool}(false)
+        # Schedule some work that's known to be expensive, to profile it
+        workload() = @async begin
+            for _ in 1:200
+                if done[] return end
+                global a = [[] for i in 1:1000]
+                yield()  # yield to allow the tests to run
+            end
+        end
+
+        @testset "task_backtraces endpoint" begin
+            @static if VERSION >= v"1.10.0-DEV.0"
+                done[] = false
+                t = workload()
+                req = HTTP.get("$url/task_backtraces", retry=false, status_exception=false)
```

Comment:
I don't understand why we are using the `done` variable like this. I'd expect the `workload` to be a while loop and that we'd set `done` to `true` after we get the response, to make the setup less racy.